controller/api/destination/watcher/endpoints_watcher.go

```diff
@@ -120,6 +132,7 @@ type (
 		Add(set AddressSet)
 		Remove(set AddressSet)
 		NoEndpoints(exists bool)
+		UpdateWithServer(set AddressSet)
```

Comment:
A bit unclear to me what the semantics of this function are.  Can `Add` be used instead?  In either case, the AddressSet should contain enough information to properly render the updated.  e.g. the AddressSet will contain the information about opaqueness.

Reply:
`Add` cannot be used here because `Add` takes into account the already available endpoints. The endpoints _could_ already be available, but they may have been updated by a Server to now be an opaque protocol.

I changed this to `Update(set AddressSet)` and it's only difference from `Add` is that doesn't do any of the diffing logic; it just creates a `WeightedAddr` for each endpoint and sends the update.